### DISCUSSION TITLE
"A listener indicated an asynchronous response by returning true, but the message channel closed before a response was received" with @webext-core/messaging

### DISCUSSION THREAD

**Discussion by Wizzel1** [wxt-dev/wxt#775] [Q&A] [answered]
So I have this content script where I call 

```ts
     //loading and handling item

      await sendMessage('uploadComplete', {
        platform: 'amazon',
        itemId: item.id,
      });
```

and in my background script there is a listener for this message where I want to save some information from the url and then reload the page:

```ts
  onMessage('uploadComplete', async (message) => {
    if (message.data.platform === 'amazon') {
      if (!amazonUploaderTabId) throw new Error('amazonUploaderTabId is null');
      const tab = await browser.tabs.get(amazonUploaderTabId);
      const array = tab.url?.split('/');
      const amazonDesignId = array?.filter((item) => z.string().uuid().safeParse(item).success)[0];
      await saveAmazonId(message.itemId, amazonDesignId);
      await browser.tabs.update(amazonUploaderTabId, {
        url: 'https://merch.amazon.com/designs/new',
      });
    }
  });
```
a user just said that the reload never happens and sent me this error log:
![image](https://github.com/wxt-dev/wxt/assets/43343510/be8cb872-52e9-4a31-967a-e9dfdb038519)

I haven't been able to reproduce it but it seems like this started happening after upgrading to `^0.18.10`. 

These are my message definitions: 

```ts
import type { Platform } from '@podsuite/types';
import { defineExtensionMessaging } from '@webext-core/messaging';
interface ProtocolMap {
  uploadComplete(data: { platform: Platform; itemId: string }): Promise<void>;
}

export const { sendMessage, onMessage } = defineExtensionMessaging<ProtocolMap>();
```

**Comment by aklinker1** [comment, marked as answer]
Hmm, messaging code looks fine... Weird. Not sure why that's happening.

I don't think it's related to 0.18.10, didn't change any runtime code, only some small parts of the build process.

This could be caused by the extension updating while the user was on the page, and something going wrong when the content script was invalidated.

If you can reproduce, I'll gladly look into this more.

**Reply by Wizzel1** [reply]
Thanks but I think a reproduction won't be possible because this error occurred on an authenticated page that not everyone can access. 

Guess I have to tinker around to find it. One more thing to mention is that this only happens in some cases and works fine in others so I guess this is some kind of race condition. 

Any hint on what to try is very appreciated!

**Reply by aklinker1** [reply]
Try reloading the extension while the authenticated page is open. Maybe that will trigger it? That's all I've got though :/

**Reply by Wizzel1** [reply]
@aklinker1 Thanks. I did some debugging together with my user and it turned out that my extension runs without any problems when he disables all other extensions. 

We are currently enabling extension by extension to see which other extension causes this but do you know any best practices to avoid problems with other extensions?

**Reply by aklinker1** [reply]
Oof, yeah, I have no idea why another extension would be interfering with yours... Sorry 😟

**Reply by Wizzel1** [reply]
No worries, I am positive that we will find the extension that causes the error. At least I know that there is nothing wrong with my extension. 😅
